analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/EmptyCollectionsShouldNotBeEnumeratedBase.cs

```diff
@@ -138,6 +138,11 @@ protected override ProgramState PreProcessSimple(SymbolicContext context)
         return state;
     }
 
+    private static bool IsEmptyObject(IOperation operation) =>
+        operation.AsObjectCreation() is { } objectCreation
+        && objectCreation.Type.IsAny(TrackedCollectionTypes)
+        && objectCreation.Constructor.Parameters.All(x => !x.Type.DerivesOrImplements(KnownType.System_Collections_IEnumerable));
```

Comment:
Why `All`? This should look at 1st argument type only, if present. Shouldn't it? Is there a constructor overload that accepts the argument at 2nd place?